src/app/admin/collectionCentresTable/CollectionCentresTable.tsx

```diff
@@ -41,49 +45,73 @@ const filters: Filter<CollectionCentresTableRow, string>[] = [
     }),
 ];
 
-interface Props {
-    collectionCentreData: CollectionCentresTableRow[];
-}
-
-const CollectionCentresTables: React.FC<Props> = (props) => {
-    const [collectionCentres, setCollectionCentres] = useState<Schema["collection_centres"][]>(
-        props.collectionCentreData
-    );
+const CollectionCentresTables: React.FC = () => {
+    const [collectionCentres, setCollectionCentres] = useState<Schema["collection_centres"][]>([]);
     const [primaryFilters, setPrimaryFilters] =
         useState<Filter<CollectionCentresTableRow, string>[]>(filters);
+    const [errorMessage, setErrorMessage] = useState<string | null>(null);
+
+    const fetchAndDisplayCollectionCentres = useCallback(async () => {
+        const { data, error } = await supabase.from("collection_centres").select();
+
+        if (error) {
+            const logId = await logErrorReturnLogId("Error with fetch: Collection Centres", {
+                error: error,
+            });
+            setErrorMessage(`Failed to fetch collection centres. Log ID: ${logId}`);
+            return;
+        }
+
+        setCollectionCentres(data);
+    }, []);
+
+    useEffect(() => {
+        void fetchAndDisplayCollectionCentres();
+    }, [fetchAndDisplayCollectionCentres]);
 
     useEffect(() => {
-        setCollectionCentres(
-            props.collectionCentreData.filter((row) => {
-                return primaryFilters.every((filter) => {
-                    return (
-                        filter.methodConfig.paginationType === PaginationType.Client &&
-                        filter.methodConfig.method(row, filter.state, filter.key)
-                    );
-                });
-            })
-        );
-    }, [primaryFilters, props.collectionCentreData]);
+        void fetchAndDisplayCollectionCentres();
+        // This requires that the DB clients, collection_centres, and families tables have Realtime turned on
+        const subscriptionChannel = supabase
+            .channel("collection-centres-table-changes")
+            .on(
+                "postgres_changes",
+                { event: "*", schema: "public", table: "collection_centres" },
+                fetchAndDisplayCollectionCentres
+            )
+            .subscribe((status, err) => {
+                if (subscriptionStatusRequiresErrorMessage(status, err, "collection_centres")) {
+                    setErrorMessage("Error fetching data, please reload");
+                }
+            });
+
+        return () => {
+            void supabase.removeChannel(subscriptionChannel);
+        };
+    }, [fetchAndDisplayCollectionCentres, primaryFilters]);
```

Comment:
I think this useeffect is just subscribing to db changes right? Why does this need to run when filters change?
```suggestion
    }, [fetchAndDisplayCollectionCentres]);
```